src/knex-adapter.js

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-underscore-dangle */
+
+const util = require('util');
+
+const dataApiClient = require('./data-api/data-api-client');
+const DataAPITransaction = require('./data-api/data-api-transaction');
+const sqlstring = require('./helper/sqlstring');
+const types = require('./helper/types');
+
+function knexAdapter(ClientRDSDataAPI, Client, dialect) {
+  // Object.setPrototypeOf(ClientRDSDataAPI.prototype, Client.prototype);
+  util.inherits(ClientRDSDataAPI, Client);
+
+  // Add/change prototype functions and properties
+  Object.assign(ClientRDSDataAPI.prototype, {
```

Comment:
Why not use `class` syntax? Or is this code mostly copied?

Reply:
This is mainly copied over. Would've been ideal to inherit `PostgresClient` class and build it. Maybe I'll keep it for next phase since it might need changing a lot?